src/core/pointcloud/qgsremotecopcpointcloudindex.cpp

```diff
@@ -111,11 +118,18 @@ std::unique_ptr<QgsPointCloudBlock> QgsRemoteCopcPointCloudIndex::nodeData( cons
     QgsDebugError( QStringLiteral( "Error downloading node %1 data, error : %2 " ).arg( n.toString(), blockRequest->errorStr() ) );
   }
 
+  storeNodeDataToCache( block.get(), n, request );
```

Comment:
looks like we're storing to cache even if the block failed to download - is that ok?

Reply:
It doesn't make sense to cache them, especially the remote ones.
I've added a check within `storeNodeDataToCacheStatic` to handle all cases.